File: 27. Remove Element/solution_2.cpp
class Solution 
{
    public:
        int removeElement(vector<int>& nums, int val) 
        {
            vector<int> newNums;
            int find = 0;

            for (int n : nums)
                if (n != val) newNums.push_back(n);
                else find++;

            nums.assign(newNums.begin(), newNums.end());

            for (int i = 0; i < find; ++i)
                nums.push_back(val);

            return size(nums) - find;
        }
};

File: leetcode/10-99/27. Remove Element/solution_2.java
class Solution 
{
    public int removeElement(int[] nums, int val) 
    {
        List<Integer> newNums = new ArrayList<>();
        int find = 0;
        
        for (int n : nums)
            if (n != val) newNums.add(n);
            else find++;
            
        for (int i = 0; i < newNums.size(); ++i)
            nums[i] = newNums.get(i);
        
        return nums.length - find;
    }
}